Source/WebKit/NetworkProcess/cache/NetworkCacheStorage.cpp

```diff
@@ -885,54 +917,63 @@ bool Storage::shouldStoreBodyAsBlob(const Data& bodyData)
 
 void Storage::dispatchWriteOperation(std::unique_ptr<WriteOperation> writeOperationPtr)
 {
-    ASSERT(RunLoop::isMain());
+    assertIsMainThread();
 
     auto& writeOperation = *writeOperationPtr;
-    m_activeWriteOperations.add(WTFMove(writeOperationPtr));
+    auto identifier = writeOperation.identifier();
+    m_activeWriteOperations.add(identifier, WTFMove(writeOperationPtr));
 
+    auto record = writeOperation.record();
```

Comment:
This is a copy as well as the copy in the capture list of the lambda at L930. We also have the original copy of the write operation stored in the `m_activeWriteOperations` array. Seems like this could lead to some duplication of Blob, although perhaps this is not too bad if the blobs are backed by memory mapped files.

It seems like we can definitely get rid of the copy in the lambda capture list (e.g. `record = crossThreadCopy(WTFMove(record))`). I'm not sure about the other copy in `m_activeWriteOperations` without more surgery.

Reply:
Or maybe just `const auto& record = writeOperation.record()` here and leave the copy in the lambda below alone.

Reply:
Although for this to do anything I guess you would also need to implement `Record::isolatedCopy() &&` which doesn't seem to currently be implemented.

Reply:
Ok, added `Record::isolatedCopy() &&`.